lib/dynamoid/persistence.rb

```diff
@@ -14,7 +14,7 @@ module Persistence
     module ClassMethods
 
       def table_name
-        @table_name ||= "#{Dynamoid::Config.namespace}_#{options[:name] || base_class.name.split('::').last.downcase.pluralize}"
+        @table_name ||= [Dynamoid::Config.namespace, options[:name] || base_class.name.split('::').last.downcase.pluralize].reject { |i| i.nil? || i.blank? }.join("_")
```

Comment:
Minor nit, all nil things are blank things, iirc, so the extra check is redundant.